packages/experience-builder-sdk/src/blocks/VisualEditorContext.tsx

```diff
@@ -272,7 +267,11 @@ export function VisualEditorContextProvider({
       window.removeEventListener('scroll', onScroll);
       clearTimeout(timeoutId);
     };
-  }, [mode, selectedNodeId]);
+  }, [mode]);
+
+  const setSelectedNodeId = (nodeId: string) => {
```

Comment:
Just want to highlight that this means that changing the selected node doesn't enforce a rerender. So if there is a case where we solely change the selected node but nothing else, we might fail updating subsequent components or UI.